raiden/smart_contracts/SecretRegistry.sol

```diff
@@ -0,0 +1,32 @@
+pragma solidity ^0.4.17;
+
+contract SecretRegistry {
+
+    /*
+     *  Data structures
+     */
+
+    string constant public contract_version = "0.3._";
+
+    // Token address => TokenNetwork address
+    mapping(bytes32 => uint64) public secret_to_block;
+
+    /*
+     *  Events
+     */
+
+    event SecretRevealed(bytes32 secret);
+
+    function registerSecret(bytes32 secret) public returns (bool) {
+        if (secret_to_block[secret] > 0) {
+            return false;
+        }
+        secret_to_block[secret] = uint64(block.number);
+        SecretRevealed(secret);
+        return true;
+    }
+
+    function getSecretBlockHeight(bytes32 secret) public constant returns (uint64) {
```

Comment:
Maybe we should upcast the block number? My idea is to match the `block.number` type.

Reply:
So you want it `uint256`?

Reply:
depends, does solidity have automatic casting? if no, I think it makes sense.

Reply:
I will note this and see if there are any storage gas cost optimizations from `uint64`. This is the main reason to downcast, but I am not sure about how mappings are stored. I'll let you know about the result. I've opened a separate issue for this : https://github.com/raiden-network/raiden-contracts/issues/2